influent-transport/src/main/java/influent/internal/nio/NioSslAcceptor.java

```diff
@@ -0,0 +1,215 @@
+      serverSocketChannel.setOption(StandardSocketOptions.SO_REUSEADDR, true);
+      serverSocketChannel.bind(this.localAddress, backlog);
+      if (receiveBufferSize > 0) {
+        serverSocketChannel
+            .setOption(StandardSocketOptions.SO_RCVBUF, receiveBufferSize);
+      }
+      eventLoop.register(serverSocketChannel, SelectionKey.OP_ACCEPT, this);
+      logger.info("A NioSslAcceptor is bound with {}.", localAddress);
+    } catch (final AlreadyBoundException | UnsupportedAddressTypeException e) {
+      throw new IllegalArgumentException(e);
+    } catch (final ClosedChannelException
+        | SecurityException
+        | UnsupportedOperationException
+        | IllegalArgumentException e) {
+      throw new AssertionError(e);
+    } catch (final IOException e) {
+      final String message = "An IO error occurred. local address = " + this.localAddress;
+      throw new InfluentIOException(message, e);
+    }
+  }
+
+  @Override
+  public void onAcceptable(final SelectionKey key) {
+    logger.debug("New connection request.");
+    while (true) {
+      SocketChannel channel = accept(key);
+      if (channel == null) {
+        break;
+      }
+      callback.accept(channel);
+    }
+  }
+
+  @Override
+  public void close() {
+    Exceptions.ignore(serverSocketChannel::close,
+        "The acceptor bound with " + localAddress + " closed.");
+    logger.info("The acceptor bound with {} closed.", localAddress);
+  }
+
+  private SocketChannel accept(SelectionKey key) {
+    logger.debug("New connection request.");
+    logger.info("New connection request.");
+    try {
+      SocketChannel channel = ((ServerSocketChannel) key.channel()).accept();
+      channel.configureBlocking(false);
+
+      SSLEngine engine = context.createSSLEngine();
+      engine.setUseClientMode(false);
+      engine.beginHandshake();
+
+      if (doHandshake(channel, engine)) {
+        return channel;
+      } else {
+        channel.close();
+        logger.debug("Connection closed due to handshake failure.");
+        logger.info("Connection closed due to handshake failure.");
+      }
+    } catch (IOException e) {
+      logger.error(e.getLocalizedMessage());
+      e.printStackTrace();
+      throw new InfluentIOException(e.getMessage(), e);
+    }
+    return null;
+  }
+
+  private boolean doHandshake(SocketChannel socketChannel, SSLEngine engine) throws IOException {
+    logger.debug("Handshaking...");
+    logger.info("Handshaking...");
+    SSLEngineResult result;
+    SSLEngineResult.HandshakeStatus status;
+    int bufferSize = engine.getSession().getApplicationBufferSize();
+    ByteBuffer buffer = ByteBuffer.allocate(bufferSize);
+    ByteBuffer unwrappedBuffer = ByteBuffer.allocate(bufferSize);
+    ByteBuffer wrappedBuffer = ByteBuffer.allocate(bufferSize);
+
+    status = engine.getHandshakeStatus();
+    logger.info("{}", status);
+    while (status != SSLEngineResult.HandshakeStatus.FINISHED &&
```

Comment:
I had thought that NioSslChannel can handle handshakes transparently, but that may be a too heavy task……